larpandoracontent/LArTwoDReco/LArClusterAssociation/MvaLowEClusterMergingAlgorithm.cc

```diff
@@ -0,0 +1,750 @@
+        const unsigned long int clusterNEdgeHits{clusterEdgeHits.size()};
+
+	std::cout << "Running 1st Loop" << std::endl;
+	for (auto iter2 = std::next(iter) ; iter2 != pClusterList->end() ; ++iter2)
+	{
+	    const Cluster *const otherCluster(*iter2);
+	    const MCParticle *otherClusterMC(this->GetMCForCluster(otherCluster, clusterToMCParticleMap));
+            sigBac = 0;
+
+            if (clusterMC == otherClusterMC)
+            {
+                   sigBac = 1;
+
+            }
+
+	    std::cout << "********* CLUSTER MC ********" << std::endl;
+            std::cout << "Cluster 1: " << clusterMC <<std::endl;
+            std::cout << "Cluster 2: " << otherClusterMC << std::endl;
+            std::cout << "Bool: " << sigBac << std::endl;    
+	    std::cout << "*****************************" << std::endl;
+
+	    CaloHitList otherClusterEdgeHits;
+	    CartesianVector centroid21(0.f, 0.f, 0.f);
+	    CartesianVector centroid22(0.f, 0.f, 0.f);
+	    try {
+	    centroid21 = otherCluster->GetCentroid(otherCluster->GetInnerPseudoLayer());
+            centroid22 = otherCluster->GetCentroid(otherCluster->GetOuterPseudoLayer());
+	    }
+	    catch (...) {
+	    CaloHitList hits;
+            cluster->GetOrderedCaloHitList().FillCaloHitList(hits);
+            centroid21 = hits.front()->GetPositionVector();
+            centroid22 = hits.front()->GetPositionVector();
+	    }
+	    const CartesianVector centroid2((centroid21 + centroid22) * 0.5f);
+	    const float centroidSeparation{Distance(centroid1, centroid2)};
+	    const CartesianVector centroidVector(centroid2.GetX() - centroid1.GetX(), 0.f, centroid2.GetZ() - centroid1.GetZ());
+	    
+	    std::string vertexListName("NeutrinoVertices3D");
+	    const VertexList *pVertexList = nullptr;
+            float vtxClusterAngle{-1};
+	    PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_INITIALIZED, !=, PandoraContentApi::GetList(*this,                       vertexListName, pVertexList));
+	    
+	    if (pVertexList && !pVertexList->empty())
+            {
+                const Vertex* pVertex{nullptr}; 
+   	        pVertex = pVertexList->front();
+	        const HitType hitType(LArClusterHelper::GetClusterHitType(cluster));
+                CartesianVector vertexPosition(0.f, 0.f, 0.f);
+
+		vertexPosition = LArGeometryHelper::ProjectPosition(this->GetPandora(), pVertex->GetPosition() , hitType);
+		const CartesianVector vtxToCluster(centroid1.GetX() - vertexPosition.GetX(), 0.f, centroid1.GetZ() - vertexPosition.GetZ());
+                const CartesianVector vtxToOtherCluster(centroid2.GetX() - vertexPosition.GetX(), 0.f, centroid2.GetZ() - vertexPosition.GetZ());
+		const float D1(std::sqrt((vtxToCluster.GetX() * vtxToCluster.GetX()) + (vtxToCluster.GetZ() * vtxToCluster.GetZ())));
+		const float D2(std::sqrt((vtxToOtherCluster.GetX() * vtxToOtherCluster.GetX()) + (vtxToOtherCluster.GetZ() * vtxToOtherCluster.GetZ())));
```

Comment:
As above, CartesianVector supports these operations directly with vec1 - vec2 and GetMagnitude